scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/SynchronousSchedulerAgent.java

```diff
@@ -445,26 +445,20 @@ private ScheduleResults collectResults(final Plan plan, final Map<ActivityInstan
           final var goalResult = new ScheduleResults.GoalResult(
               goalEval
                   .getValue()
-                  .getInsertedActivities().stream().map(activityInstance -> {
-                    ActivityInstanceId id;
-                    if(activityInstance.isGenerated()){
-                      id = instancesToIds.get(activitiesById.get(activityInstance.getParentActivity()));
-                    } else{
-                      id = instancesToIds.get(activityInstance);
-                    }
-                    return id;
-                  }).toList(),
+                  .getInsertedActivities().stream()
+                      .map(activityInstance -> instancesToIds.get(
+                          activityInstance.getParentActivity()
+                              .map(activitiesById::get)
+                              .orElse(activityInstance))
+                  ).toList(),
               goalEval
                   .getValue()
-                  .getAssociatedActivities().stream().map(activityInstance -> {
-                    ActivityInstanceId id;
-                    if(activityInstance.isGenerated()){
-                      id = instancesToIds.get(activitiesById.get(activityInstance.getParentActivity()));
-                    } else{
-                      id = instancesToIds.get(activityInstance);
-                    }
-                    return id;
-                  }).toList(),
+                  .getAssociatedActivities().stream()
+                      .map(activityInstance -> instancesToIds.get(
+                        activityInstance.getParentActivity()
+                            .map(activitiesById::get)
+                            .orElse(activityInstance))
+                    ).toList(),
```

Comment:
Just double-checking, the parenthesis is closing the get on line 457, right?

Reply:
Oops, the line-breaking and parentheses-breaking here is formatted kind of strangely. That final parenthesis is for the `map`. Just pushed formatting change for this :)